### PR TITLE
Use ReTest

### LINKED ISSUE #83: Use ReTest or similar niceness?
FASTX is composed of the FASTA and FASTQ modules, and a little bit of common code.
After the test refactoring for v2 (see #37 , implemented in #68 ), it should be relatively straightforward to use shiny new test toys like ReTest so that new features could be selectively tested.

CC @kescobo , who recently showed interest in applying ReTest to Automa, IIRC :)

### PR DESCRIPTION
**Pull request by kescobo**
Branched off of #68 . Closes #83

### CODE REVIEW

**Comment by kescobo** [comment]
@jakobnissen [So this is a bummer](https://juliatesting.github.io/ReTest.jl/stable/#Including-files-from-within-testsets)

How married to the nesting testsets are you? I could preserve it by making each of the sub-files a new module (eg TestFASTA.IO - it would be a bit of a hassle since each thing would need to re-do imports etc. Another option is to just include the name in all of the testsets (eg `@testset "IO - Reader Basics" begin`. Then one can do `retest(TestFASTA, "IO") to run them in isolation.

I'm partial to the last one. You can pass multiple filters, so eg if you want the "basic" tests, but only the ones from "IO", you can do `retest(TestFASTA, "IO", "basic")`.

**Comment by kescobo** [comment]
Argh, actually, that doesn't work as well because it doesn't pick up on changes to the `include`d files with Revise. I'll just make submodules.

**Comment by jakobnissen** [comment]
I don't really have any opinions, since I haven't used ReTest for any real project yet.

**Comment by kescobo** [comment]
Alright, I think this should work. I've added a `test/README.md` to describe what to do, please take a look and let me know if it makes sense.

**Comment by kescobo** [comment]
Hm... don't see why julia 1.6 would be different here.

**Comment by kescobo** [comment]
Oh, the problem is windows/mac, not the julia version :thinking: 

I only have linux box, anyone with mac/windows have a thought

**Comment by kescobo** [comment]
@jakobnissen OK, I think this is good to go

**Review by jakobnissen — CHANGES_REQUESTED** [review]
Real nice.

File: src/FASTX.jl [hunk 1]
```diff
@@ -4,6 +4,13 @@ using StringViews: StringView
 using BioSequences: BioSequence, LongSequence
 using Automa: Automa
 
+# using ReTest
+#
+# @testset "Inline test placeholder" begin
+#     @test false
+# end
+
+
```
Comment (jakobnissen) [inline src/FASTX.jl:13]:
```suggestion
```

File: test/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+❯ julia --project
+               _
+   _       _ _(_)_     |  Documentation: https://docs.julialang.org
+  (_)     | (_) (_)    |
+   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
+  | | | | | | |/ _` |  |
+  | | |_| | | | (_| |  |  Version 1.7.3 (2022-05-06)
+ _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
+|__/                   |
+
```
Comment (jakobnissen) [inline test/README.md:19]:
```suggestion
❯ julia --project -q
```

**Review by jakobnissen — APPROVED** [review]
Real nice.

**Pull request merged**